sycl/test-e2e/Matrix/joint_matrix_out_bounds_impl.hpp

```diff
@@ -0,0 +1,155 @@
+#define TM 8
+#define TN SG_SZ
+#define TK 16
+
+#define BF16_EPSILON 0.00781250
+
+template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
+public:
+  T *mat;
+
+public:
+  T *get_data() { return mat; }
+  void set_data(T *data) { mat = data; }
+  big_matrix(T *data) : mat(data) {}
+};
+
+float make_fp32(bfloat16 x) {
+  unsigned int y = *((int *)&x);
+  y = y << 16;
+  float *res = reinterpret_cast<float *>(&y);
+  return *res;
+}
+
+template <typename T1, typename T2, size_t NUM_ROWS_A, size_t NUM_COLS_A,
+          size_t NUM_ROWS_B, size_t NUM_COLS_B, size_t NUM_ROWS_C,
+          size_t NUM_COLS_C>
+void matrix_multiply(big_matrix<T1, NUM_ROWS_C, NUM_COLS_C> &C,
```

Comment:
It seems this code is generally a duplicate of other matrix multiply kernels in this test repo except some changes in ND range and matrix M size. Is it possible to avoid duplication? Maybe putting common code (like matrix_multiply kernel) with necessary parameters to a separate file and in this new test we would just set parameters that are differ from other similar tests?

Reply:
I would prefer to keep the main kernel self contained in one function to make easy understanding joint matrix usage.